examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -707,46 +699,48 @@ void ProcessEvent(WfxEvent_t inEvent)
 #ifdef SL_WFX_CONFIG_SCAN
         rsi_rsp_scan_t scan_rsp = { 0 };
-        int32_t status          = rsi_wlan_bgscan_profile(1, &scan_rsp, sizeof(scan_rsp));
+        memset(&scan_rsp, 0, sizeof(scan_rsp));
+        int32_t status = rsi_wlan_bgscan_profile(1, &scan_rsp, sizeof(scan_rsp));
 
-        if (status)
+        if (status != RSI_SUCCESS)
         {
-            SILABS_LOG("SSID scan failed: %02x ", status);
+            SILABS_LOG("rsi_wlan_bgscan failed: %02x ", status);
+            return;
         }
-        else
+        rsi_scan_info_t * scan;
+        wfx_wifi_scan_result_t ap;
+        for (int x = 0; x < scan_rsp.scan_count[0]; x++)
         {
-            rsi_scan_info_t * scan;
-            wfx_wifi_scan_result_t ap;
-            for (int x = 0; x < scan_rsp.scan_count[0]; x++)
+            scan = &scan_rsp.scan_info[x];
+            // is it a scan all or target scan
+            if (!wfx_rsi.scan_ssid || (wfx_rsi.scan_ssid && strcmp(wfx_rsi.scan_ssid, (char *) scan->ssid) == CMP_SUCCESS))
             {
-                scan = &scan_rsp.scan_info[x];
-                // is it a scan all or target scan
-                if (!wfx_rsi.scan_ssid || (wfx_rsi.scan_ssid && strcmp(wfx_rsi.scan_ssid, (char *) scan->ssid) == CMP_SUCCESS))
+                // clear structure and calculate size of SSID
+                memset(&ap, 0, sizeof(ap));
+                strncpy(ap.ssid, (char *) scan->ssid, strnlen((const char *) scan->ssid, WFX_MAX_SSID_LENGTH));
+                ap.security = scan->security_mode;
+                ap.rssi     = (-1) * scan->rssi_val;
+
+                configASSERT(sizeof(ap.bssid) == BSSID_LEN);
+                configASSERT(sizeof(scan->bssid) == BSSID_LEN);
+                memcpy(ap.bssid, scan->bssid, BSSID_LEN);
+                (*wfx_rsi.scan_cb)(&ap);
+
+                if (wfx_rsi.scan_ssid)
                 {
-                    strncpy(ap.ssid, (char *) scan->ssid, MIN(sizeof(ap.ssid), sizeof(scan->ssid)));
-                    ap.security = scan->security_mode;
-                    ap.rssi     = (-1) * scan->rssi_val;
-                    configASSERT(sizeof(ap.bssid) >= BSSID_LEN);
-                    configASSERT(sizeof(scan->bssid) >= BSSID_LEN);
-                    memcpy(ap.bssid, scan->bssid, BSSID_LEN);
-                    (*wfx_rsi.scan_cb)(&ap);
-
-                    if (wfx_rsi.scan_ssid)
-                    {
-                        break; // we found the targeted ssid.
-                    }
+                    break; // we found the targeted ssid.
                 }
             }
         }
 
         /* Terminate with end of scan which is no ap sent back */
-        (*wfx_rsi.scan_cb)((wfx_wifi_scan_result_t *) 0);
-        wfx_rsi.scan_cb = (void (*)(wfx_wifi_scan_result_t *)) 0;
+        (*wfx_rsi.scan_cb)((wfx_wifi_scan_result_t *) NULL);
+        wfx_rsi.scan_cb = NULL;
 
         if (wfx_rsi.scan_ssid)
         {
             vPortFree(wfx_rsi.scan_ssid);
```

Comment:
This is allocated with ChipMemoryAlloc on line 349 (`    wfx_rsi.scan_ssid = reinterpret_cast<char *>(chip::Platform::MemoryAlloc(ssid_len + 1));`)

You cannot use vPortFree since the MemoryAlloc may have hit a different allocator than FreeRTOS

Please don't use vPortXxxxx for heap allocation, only chip::platform::MemoryXxxx pairs, or, if all platform code, `calloc/free` pairs directly.